libhb/qsv_common.c

```diff
@@ -2183,7 +2183,7 @@ static int hb_qsv_parse_options(hb_job_t *job)
 int hb_qsv_setup_job(hb_job_t *job)
 {
     // parse the json parameter
-    if (job->qsv.ctx && job->qsv.ctx->dx_index >= 0)
+    if (job->qsv.ctx && job->qsv.ctx->dx_index >= -1)
```

Comment:
HW configuration:
other HW vendor - index 0
Intel iGPU - index 1

Previous version failed in GUI binary if Intel GPU has no index 0. In GUI this if statement is always false since job->qsv.ctx->dx_index=-1. As result decoder initialized with index 0 which is not QSV device.

In console app this is different since command line parameters parsed.